docs/CODE_GUIDELINES.md

```diff
@@ -764,4 +862,44 @@ Try to avoid using C macros. In many cases, they can be easily substituted with
 
 Prefer `constexpr` over `const` for constants when possible. Try to mark functions `constexpr` when reasonable.
 
+### 12.11. `std::string` vs `std::string_view`
+
+Prefer `std::string_view` over `std::string` when reasonable. Good examples are constants or method arguments. In the last case, it is not required declare the argument as reference or const, since the data source of string views are immutable by definition. A bad example is when you need a NUL-terminated C string, e.g. to interact with a C API. `std::string_view` does not offer a `c_str()` function like `std::string` does, but if you do not need a C string you can use `data()` to get the raw source of the data.
+
+Main reasons why we prefer `std::string_view` are: execution performance, no memory allocations, substrings can be made without copy, and the possibility to reuse the same data without reallocation.
+
+Avoid using `std::string_view` when you are not sure where the source of data is allocated, or as return value of a method. If not handled properly, the source (storage) of the data may go out of scope. As a consequence, the program enters undefined behavior and may crash, behave strangely, or exhibit security-relevant issues.
```

Comment:
_(Question - "or exhibit security-relevant issues" seems an odd way to phrase it, but this may be deliberate. Would not "or introduce potential security issues" be clearer?)_